zcash_client_backend/proto/proposal.proto

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+syntax = "proto3";
+package cash.z.wallet.sdk.ffi;
+
+// A data structure that describes the inputs to be consumed and outputs to
+// be produced in a proposed transaction.
+message Proposal {
+    uint32 protoVersion = 1;
+    // ZIP 321 serialized transaction request
+    string transactionRequest = 2;
+    // The unique identifier and amount for each proposed input
+    repeated ProposedInput inputs = 3;
+    // The total value, fee amount, and change outputs of the proposed
+    // transaction
+    TransactionBalance balance = 4;
+    // The fee rule used in constructing this proposal
+    FeeRule feeRule = 6;
+    // The minimum number of confirmations required for transaction inputs to be selected.
+    // Each of the proposed inputs is know to have at least this many confirmations.
+    uint32 minConfirmations = 7;
+    // The target height for which the proposal was constructed
+    //
+    // The chain must contain at least this many blocks in order for the proposal to
+    // be executed.
+    uint32 minTargetHeight = 8;
+    // Returns the anchor height used in preparing the proposal.
+    //
+    // If, at the time that the proposal is executed, the anchor height
+    // required to satisfy / the minimum confirmation depth is less than this
+    // height, the proposal execution / API will return an error.
+    uint32 minAnchorHeight = 9;
+    // A flag indicating whether the proposal is for a shielding transaction,
+    // used for determining which OVK to select for wallet-internal outputs.
+    bool isShielding = 10; 
+}
+
+// The value pool from which a particular proposed input is drawn
+enum ValuePool {
+    Transparent = 0;
+    Sapling = 2;
+}
+
+// The unique identifier, value pool, and amount for each proposed input
+message ProposedInput {
+    bytes txid = 1;
+    ValuePool pool = 2;
+    uint32 index = 3;
+    int64 value = 4;
+}
+
+// The fee rule used in constructing a Proposal
+enum FeeRule {
+    // 10000 ZAT
+    PreZip313 = 0;
```

Comment:
Use `Auto` for discriminant 0, so that if the proposer doesn't specify a fee rule, the "default" becomes "use whatever the wallet believes is the best fee rule".

Reply:
Hmm, except that this then leaves the change and fee fields overconstrained. Instead let's call this `Unspecified`, and then the proposal consumer can treat it as an invalid state.